File: backend/app/Models/PurchaseReturn.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class PurchaseReturn extends Model
{
    use HasFactory;

    /**
     * The attributes that are mass assignable.
     *
     * @var array<int, string>
     */
    // تأكد تمامًا أن هذا الجزء مطابق
    protected $fillable = [
        'purchase_invoice_id',
        'user_id',
        'date',      // <-- هذا هو السطر الناقص
        'total',
        'reason',
        'created_by',
    ];

    

    public function purchaseInvoice()
    {
        return $this->belongsTo(PurchaseInvoice::class);
    }

    public function items()
    {
        return $this->hasMany(PurchaseReturnItem::class);
    }

    public function user()
    {
        return $this->belongsTo(User::class, 'user_id');
    }

    public function creator()
    {
        return $this->belongsTo(User::class, 'created_by');
    }
}

File: desktop/lib/features/purchases/domain/entities/purchase_return.dart
import 'package:desktop/features/purchases/domain/entities/purchase_return_item.dart';

class PurchaseReturn {
  final String id;
  final String purchaseInvoiceId;
  final String? supplierName;
  final int userId;
  final String date;
  final double total;
  final String? reason;
  final int createdBy;
  final List<PurchaseReturnItem> items;
  final bool isSynced;
  final String? createdAt;

  const PurchaseReturn({
    required this.id,
    required this.purchaseInvoiceId,
    this.supplierName,
    required this.userId,
    required this.date,
    required this.total,
    this.reason,
    required this.createdBy,
    required this.items,
    this.isSynced = false,
    this.createdAt,
  });
}
